src/__tests__/extensions/replay/sessionrecording.test.ts

```diff
@@ -2156,4 +2157,77 @@ describe('SessionRecording', () => {
             )
         })
     })
+
+    describe('URL blocking', () => {
+        beforeEach(() => {
+            sessionRecording.startIfEnabledOrStop()
+            sessionRecording.afterDecideResponse(
+                makeDecideResponse({
+                    sessionRecording: {
+                        endpoint: '/s/',
+                        urlBlocklist: [
+                            {
+                                matching: 'regex',
+                                url: '/blocked',
+                            },
+                        ],
+                    },
+                })
+            )
+        })
+
+        it('flushes buffer and includes pause event when hitting blocked URL', async () => {
+            // Emit some events before hitting blocked URL
+            _emit(createIncrementalSnapshot({ data: { source: 1 } }))
+            _emit(createIncrementalSnapshot({ data: { source: 2 } }))
+
+            // Simulate URL change to blocked URL
+            fakeNavigateTo('https://test.com/blocked')
+            _emit(createIncrementalSnapshot({ data: { source: 3 } }))
+
+            await waitFor(() => {
+                // Verify the buffer was flushed with all events including pause
+                expect(posthog.capture).toHaveBeenCalledWith(
+                    '$snapshot',
+                    {
+                        $session_id: sessionId,
+                        $window_id: 'windowId',
+                        $snapshot_bytes: expect.any(Number),
+                        $snapshot_data: [
+                            { type: 3, data: { source: 1 } },
+                            { type: 3, data: { source: 2 } },
+                            // This should be in the buffer
+                            // { type: 5, data: { tag: 'recording paused', payload: { reason: 'url blocker' } } },
```

Comment:
ah, yes, this'll be because mocked rrweb i bet... may well be a way we can improve the mock to pass along custom events - although feel free to not spend too much time on that

if I really care then i put tests into cypress so i'm exercising a real rrweb